openmeetings-core/src/main/java/org/apache/openmeetings/core/remote/StreamProcessor.java

```diff
@@ -494,20 +498,24 @@ void remove(Client c) {
 			}
 		}
 		if (c.getRoomId() != null) {
-			KRoom room = kHandler.getRoom(c.getRoomId());
-			room.leave(this, c);
 			checkStreams(c.getRoomId());
 		}
 	}
 
-	void addStream(KStream stream) {
+	public void addStream(KStream stream) {
 		streamByUid.put(stream.getUid(), stream);
 	}
 
 	public Collection<KStream> getStreams() {
 		return streamByUid.values();
 	}
 
+	public Collection<KStream> getStreamsByRoom(Long roomId) {
+		return streamByUid.values().stream()
+				.filter(stream -> stream.getRoom() != null && stream.getRoom().getRoomId().equals(roomId))
+				.collect(Collectors.toCollection(ArrayList::new));
```

Comment:
Why not to use `Collectors.toList()` ?

Do you have any benchmarks on how much this is slower than separate list?

Reply:
It seems to be rather the same or faster:
https://stackoverflow.com/questions/40893043/java-8-stream-list-collector-memory-allocation-speed-vs-loop-with-preallocation/40893132#40893132

Reply:
It will be faster only in case you will provide correct size in the constructor
This construct is just heavier